lib/ci/ci_result_parser.js

```diff
@@ -343,24 +354,26 @@ class TestBuild extends Job {
 }
 
 function getHighlight(f) {
-  return f.reason.split('\n')[f.highlight]
-    .replace(/not ok \d+ /, '')
-    .replace(
-      /JNLP4-connect connection from \S+/, 'JNLP4-connect connection from ...'
-    )
-    .replace(/FATAL: Could not checkout \w+/, 'FATAL: Could not checkout ...')
-    .replace(
-      /error: pathspec .+ did not match any file\(s\) known to git/,
-      'error: pathspec ... did not match any file(s) known to git')
-    .replace(
-      /failed: no workspace for .+/,
-      'failed: no workspace for ...'
-    )
-    .replace(
-      /fatal: loose object \w+ \(stored in .git\/objects\/.+\) is corrupt/,
-      'fatal: loose object ... (stored in .git/objects/...) is corrupt')
-    .replace(/hudson\.plugins\.git\.GitException: /, '')
-    .replace(/java\.io\.IOException: /, '');
+  if (f.reason != undefined){
+    return f.reason.split('\n')[f.highlight]
+      .replace(/not ok \d+ /, '')
+      .replace(
+        /JNLP4-connect connection from \S+/, 'JNLP4-connect connection from ...'
+      )
+      .replace(/FATAL: Could not checkout \w+/, 'FATAL: Could not checkout ...')
+      .replace(
+        /error: pathspec .+ did not match any file\(s\) known to git/,
+        'error: pathspec ... did not match any file(s) known to git')
+      .replace(
+        /failed: no workspace for .+/,
+        'failed: no workspace for ...'
+      )
+      .replace(
+        /fatal: loose object \w+ \(stored in .git\/objects\/.+\) is corrupt/,
+        'fatal: loose object ... (stored in .git/objects/...) is corrupt')
+      .replace(/hudson\.plugins\.git\.GitException: /, '')
+      .replace(/java\.io\.IOException: /, '');
+  } else return f.reason = "failure not found";
```

Comment:
you could reduce the diff, and simplify the logic, by leaving the code as it was and returning early when the exceptional condition is found:

```
if (!f.reason)
   return f.reason = 'failure not found'
```